src/S3Writer.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\S3Writer;
+
+use Aws\S3\Exception\S3Exception;
+use Aws\S3\S3Client;
+use Aws\S3\S3MultiRegionClient;
+use GuzzleHttp\Exception\ClientException;
+use Keboola\Component\UserException;
+use Monolog\Logger;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+
+class S3Writer
+{
+    /**
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * @var Logger
+     */
+    private $logger;
+
+    /**
+     * Application constructor.
+     */
+    public function __construct(Config $config, Logger $logger)
```

Comment:
`LoggerInterface`

Reply:
Mě se vlastně moc nelíbí dependency na Configu přímo. IMHO by to už mělo chtít jednotlivé ty parametry. Když bys to chtěl otestovat v izolaci, tak si musíš vyrobit zbytečně celý Config. Chápu to tak, že ten writer je reusable třída. (taky nepočítá s tím, že čte z /out/files (protože je v KBC), ale má to jako parametr.

Reply:
Taky jsem si myslel, že by to mohlo být lepší. Trochu mě na tom jen zarazilo, že potenciálně můžou argumenty konstruktoru růst donekonečna, takže pak by se to předělávalo do nějakýho konfiguračního objektu, což je vlastně ten `Config`. Jinak se to tam bude někde přehazovat z původního `Config`u do nějakýho novýho, kterej bude nachlup stejnej, což mi teda přijde ultra zbytečný.